### PR TITLE
[SELS_TASK] [BE] Deployment | Create Config using Environment Variables

### PR DESCRIPTION
**Pull request by ww-jermaine**
**[Asana Link]**
[https://app.asana.com/0/1202968674806069/1202968800627006/f](https://app.asana.com/0/1202968674806069/1202968800627006/f
)

**[Commands]**
1. Create and activate the virtual environment.
```
cd backend
python -m venv venv
venv/Scripts/Activate
```
2. Install the packages listed in `requirements.txt`
```
pip install -r requirements.txt
```

3. Make a copy of `.env.example`
```
cp .env.example .env
```

**[Pre-conditions]**
Please edit the `SECRET_KEY` entry of `.env`.

See Screenshot 1.

Let the remaining entries unfilled. Running `python manage.py check`, there should be no errors:

See Screenshot 2.

Another way to check that this is working properly is to run `python manage.py print_settings`, the command should work when `django-extensions` is installed. It is listed under `requirements.txt`.

See Screenshot 3.

**[Notes]**
The output of `python manage.py print_settings` could be lengthy, for linux I think `python manage.py print_settings | grep SECRET_KEY` should print out the entry for SECRET_KEY only.

**[Screenshots]**

Screenshot 1
image

Screenshot 2
image

Screenshot 3
image

### CODE REVIEW

File: backend/core/settings.py [hunk 1]
```diff
@@ -1,32 +1,27 @@
...
+environ.Env.read_env(os.path.join(BASE_DIR, ".env"))
 
 # Quick-start development settings - unsuitable for production
 # See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = "django-insecure-j4f5%uc0-t^@r88q%%5vs1^3e%$03(n_fx#c-cx*!c13ine)2="
+SECRET_KEY = env(
+    "SECRET_KEY",
+    default="django-insecure-)c6r+c&1r&&-d#ma(^f5k=g5mx*z256q&ipwczhf_=qqk0ze=i",
```
Comment (Alvilb) [inline backend/core/settings.py]: lets remove this default

File: backend/core/settings.py [hunk 2]
```diff
@@ -1,32 +1,27 @@
...
+)
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = True
-
-ALLOWED_HOSTS = []
+DEBUG = env.bool("DEBUG", default=True)
```
Comment (Alvilb) [inline backend/core/settings.py]: same here. lets remove the default value

File: backend/core/settings.py (hunk 1 above)
Comment (ww-jermaine) [reply backend/core/settings.py]: All right sir, removed the defaults for SECRET_KEY and DEBUG in this latest commit.
[https://github.com/framgia/sph-els-marc/pull/1/commits/6470a3ea53aa081a9e30f58438e0dff2837bf3df](https://github.com/framgia/sph-els-marc/pull/1/commits/6470a3ea53aa081a9e30f58438e0dff2837bf3df)

**Review by Alvilb — APPROVED** [review]
Approved!

**Pull request merged**